### PR TITLE
Using CodeforcesApiPy and fixed Call limit

### PR DESCRIPTION
**Pull request by VadVergasov**
Using [CodeforcesApiPy](https://github.com/VadVergasov/CodeforcesApiPy) and waiting for some time to avoid Call Limit exceeded.

### CODE REVIEW

**Comment by nileshsah** [comment]
Thanks for the proposed contribution @VadVergasov! Great to see that you already are maintaining a python wrapper for the Codeforces API. If possible, I'd like to move away from custom request calls entirely in that case by extending your project to support APIs for (1) getting tags for a given problem and (2) getting the total number of submissions page for a user. 
Let me know if you have the bandwidth to add these additional functionalities else I'd be happy to contribute as well.

File: harwest/lib/codeforces/client.py [hunk 1]
```diff
@@ -28,32 +32,18 @@ def get_submissions_page_count(self):
...
-            submission_id=submission_id
-        )
-        sub_soup = self.__get_content_soup(sub_url)
-        # For debug purpose
-        # print(sub_url)
-        # open("last_submission_page.html", "w").write(str(sub_soup))
-        submission_code = sub_soup.find('pre', attrs={'id': 'program-source-text'})
-        if submission_code is None:
+        try:
+            return self.cf_parser.get_solution(contest_id, submission_id)
```
Comment (nileshsah) [inline harwest/lib/codeforces/client.py:36]: What would happen if we call `get_solution(...)` on a submission id which is not public or unavailable?

**Comment by VadVergasov** [comment]
First of all, we need to wait for Codeforces API fixes, I've already contacted Mike Mirzayanov because if we need to get a lot of info in one request we need to use the POST method, not GET, but Codeforces responses 403 error on such requests.

(1) In Codeforces API we have [problemset.problems](https://codeforces.com/api/problemset.problems) method, but it will return the array of problems and we need to look throw the whole tasks to get their tags. So it will take some time. But for me, it takes about 0.5 seconds to get and convert the response to the dictionary with such look: `tags["contestId"]["index"] = [tags]`
(2) In Codeforces API we have [user.status](https://codeforces.com/api/user.status?handle=VadVergasov) and it will return all user's submissions, so we don't need to parse tasks page by page and this will throw away some problems, connected with a page processing (like #5 I can try to edit the current realization to modify that.

File: harwest/lib/codeforces/client.py (hunk 1 above)
Comment (VadVergasov) [reply harwest/lib/codeforces/client.py:36]: ValueError: Incorrect contest_id or submit_id. But as said in #7 we can create auth in session to get the solution. For now Codeforces API doesn't provide any tool to get solutions.

Comment (nileshsah) [reply harwest/lib/codeforces/client.py:36]: Sounds good! Would love to leverage the auth session feature sometime in the future when we decide to support user login on Harwest.

**Comment by nileshsah** [comment]
Thanks for the update @VadVergasov! I guess I forgot to update #5 but I had briefly addressed it when I rolled out the `--full-scan` feature. Also, I just recalled and checked that I am no more using the `get_submissions_page_count` method which practically means that we can drop (2) completely. Sorry for the confusion.

**Comment by VadVergasov** [comment]
Updated getting tags

**Pull request closed without merging**